test/Tests/src/Semantic/Meta_Spec.enso

```diff
@@ -390,11 +393,33 @@ spec =
             case atom.bar of
                 n : Number -> Test.fail "Not changed to number: "+n
                 _ -> Nothing
-    Test.group "Unresolved symbol"
+
+    Test.group "Unresolved symbol" <|
         Test.specify "should be treated as a Function" <|
             (_.is_nothing) . is_a Function . should_equal True
             (.is_nothing) . is_a Function . should_equal True
             Meta.type_of (_.is_nothing) . should_equal Function
             Meta.type_of (.is_nothing) . should_equal Function
 
+    Test.group "Extension methods and get_annotation" <|
+        it = Meta_Spec_Type.Value 6 7
+
+        Test.specify "verify calling everything works" <|
+            it.describe . should_equal 6
+            it.describe "second" . should_equal 7
+            Panic.catch Any (it.describe_better) (_->"Not executed") . should_equal "Not executed"
+            Meta_Define_Usage.call_describe_better it . should_equal 42
+
+        Test.specify "verify calling everything works" <|
+            describe_anno = Meta.get_annotation it "describe" "which"
+            describe_anno it . should_equal 13
```

Comment:
This is very "verbose", @jdunkerley! One has to: 
- a lambda function `@which (self -> self.a)` as annotation
- gets a lambda function via `Meta.get_annotation`
- needs to invoke it

I was hoping `Meta.get_annotation` actually invokes the code in `@which self.a` by itself internally! E.g. one could reference `self` in the code after `@which` and it evaluation would happen by itself too.

With the current state of affairs, there is no reason why `Meta.get_annotation` shall take `self` argument! It rather needs a `Function`!

Reply:
67fb583 let's anyone _"resolve"_ a function and then pass such function as a _handle_ around. Then everyone can query for annotations associated with that function.

Reply:
Agree on the `get_annotation` having to unwind to get the value is annoying.

I wrapped it in the visualization layer and haven't thought of it since!